cmd/backup/main.go

```diff
@@ -430,8 +425,15 @@ func (s *script) copyBackup() error {
 	return nil
 }
 
-// removeArtifacts removes the backup file from disk.
+// removeArtifacts removes the backup file from disk. Also remove snapshot if snapshot was created
 func (s *script) removeArtifacts() error {
+	if s.c.BackupFromSnapshot {
```

Comment:
I put remove snapshot before remove artifact, reverse to the archive creation order in `takeBackup`. This has the benefit that if creating snapshot already failed before creating archive, snapshot still get removed here.